deps/rabbit/src/rabbit_quorum_queue.erl

```diff
@@ -230,27 +230,48 @@ declare(Q, _Node) when ?amqqueue_is_quorum(Q) ->
 
 start_cluster(Q) ->
     QName = amqqueue:get_name(Q),
-    Durable = amqqueue:is_durable(Q),
-    AutoDelete = amqqueue:is_auto_delete(Q),
-    Arguments = amqqueue:get_arguments(Q),
     Opts = amqqueue:get_options(Q),
+    Arguments = amqqueue:get_arguments(Q),
     ActingUser = maps:get(user, Opts, ?UNKNOWN_USER),
     QuorumSize = get_default_quorum_initial_group_size(Arguments),
+    {LeaderNode, FollowerNodes} =
+        rabbit_queue_location:select_leader_and_followers(Q, QuorumSize),
     RaName = case qname_to_internal_name(QName) of
                  {ok, A} ->
                      A;
                  {error, {too_long, N}} ->
                      rabbit_data_coercion:to_atom(ra:new_uid(N))
              end,
-    {LeaderNode, FollowerNodes} =
-        rabbit_queue_location:select_leader_and_followers(Q, QuorumSize),
+    rabbit_log:debug("Will start up to ~w replicas for quorum ~ts with leader on node '~ts'",
+                     [QuorumSize, rabbit_misc:rs(QName), LeaderNode]),
+    case internal_declare(Q, RaName, LeaderNode, FollowerNodes) of
+        {error, cluster_not_formed, Queue} ->
+            _ = rabbit_amqqueue:internal_delete(Queue, ActingUser),
+            AlternativeRaName = rabbit_data_coercion:to_atom(ra:new_uid(rabbit_queue_type_util:name_concat(QName))),
+            rabbit_log:debug("Could not form cluster ~ts, trying alternative name ~ts",
+                     [RaName, AlternativeRaName]),
+            case internal_declare(Q, AlternativeRaName, LeaderNode, FollowerNodes) of
```

Comment:
This doesn't handle every error `internal_declare/4` can return.

Reply:
it also doesn't handle the partial case where there is a name conflict in a minority subset of the members.